docs/iam-usage.rst

```diff
@@ -0,0 +1,177 @@
+
+>>> policy, version, etag = resource.get_policy()
+>>> print(policy)
+{
+   'roles/owner': set(['user:alice@example.com']),
+   'roles/editor: set(['group:admins@example.com']),
+   'roles/reader': set(['domain:example.com', 'user:bob@example.com'])
+}
+>>> print(version)
+5
+>>> print(etag)
+ffdFADFdsgfsjrsHTY
+
+``set_policy`` takes a policy dictionary, as well as optional ``version`` and ``etag`` paramters. If updates are made to your policy during this change, they will be overwritten with exactly what is in your policy, or, if an etag is specified they will fail with a ``iam.ConcurrentModificationError``. ``iam.PolicyChange`` performs this "read-modify-write" cycle automatically for the user.
+
+>>> policy['roles/owner'].add('user:charles@example.com')
+>>> policy, version, etag = resource.set_policy(policy, etag=etag)
+>>> print(version)
+6
+
+``missing_permissions`` takes an iterable of "permission strings" and returns those the user does not have on the resource
+
+>>> resource.missing_permissions('resourcemanager.projects.get', 'resourcemanager.projects.delete')
+set(['resourcemanager.projects.get'])
+
+Returns permissions (if any), in the list that the user does not possess.
+
+``query_grantable_roles()`` returns a list of ``iam.Role`` objects that represent roles (and their associated metadata)
+which can be granted on the specified resource
+
+>>> resource.query_grantable_roles()
+[<Role>, <Role>, <Role>]
+
+Convenience Methods
+~~~~~~~~~~~~~~~~~~~
+
+The following methods are wrappers around the creation and application of an ``iam.PolicyChange`` object. 
+
+``add_role`` takes a single member, and a single ``iam.Role``, or role string, and adds the member to the role. ``add_role``
+
+>>> resource.add_role(iam.user('alice@example.com'), iam.roles.OWNER.name)
+
+``remove_role`` has the same signature as ``add_role`` but removes the member from the role.
+
+>>> resource.remove_role(iam.user('bob@example.com'), iam.roles.OWNER.name)
+
+``add_roles`` takes a single member, and an iterable of ``iam.Role`` s or role strings, and the member to each role
+
+>>> resource.add_roles(iam.user('alice@example.com'), [iam.roles.OWNER.name, iam.roles.EDITOR.name])
+
+``remove_roles`` has the same signature as ``resource.add_roles`` but removes all the specified roles from the member (where present)
+
+>>> resource.remove_roles(iam.group('devs@example.com'), [iam.roles.OWNER.name, iam.roles.EDITOR.name])
+
+``add_members`` takes an ``iam.Role`` and an iterable of members and adds each member to the role
+
+>>> resource.add_members(iam.roles.OWNER.name, [iam.domain('example.com'), iam.service_account('compute@iam.my-project.example.com')])
+
+``remove_members`` has the same signature as ``add_members`` but removes all the members from the specified role.
+
+>>> resource.remove_members(iam.roles.OWNER.name, [iam.ALL_USERS])
+
+
+IAM for Contributors
+==========================
```

Comment:
I applaud you writing this section but it should probably go somewhere else, like `CONTRIBUTING.rst`